demo/java/src/main/java/io/github/cosinekitty/astronomy/demo/LunarEclipse.java

```diff
@@ -12,8 +13,8 @@ public class LunarEclipse {
      * The date and time after which to start searching for lunar eclipses.
      */
     public static int run(Time startTime) {
-        LunarEclipseInfo e = Astronomy.searchLunarEclipse(startTime);
-        Stream.iterate(e, x -> Astronomy.nextLunarEclipse(x.getPeak()))
+        Iterator<LunarEclipseInfo> iterator = Astronomy.lunarEclipsesAfter(startTime);
+        Stream.iterate(iterator.next(), x -> iterator.next())
```

Comment:
This is kinda a hack, there are other solutions https://stackoverflow.com/q/24511052 which didn't include this one but I found enough for the use here.